src/EditorFeatures/CSharpTest/PullMemberUp/CSharpPullMemberUpTests.cs

```diff
@@ -42,6 +42,75 @@ public class TestClass : ITestInterface
             await TestNoRefactoringProvidedAsync(testText);
         }
 
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsPullMemberUp)]
+        public async Task TestNoRefactoringProvidedWhenDeclarationAlreadyExists()
+        {
+            var methodTest = @"
+namespace PushUpTest
+{
+    public interface ITestInterface
+    {
+        void TestMethod();
+    }
+
+    public class TestClass : ITestInterface
+    {
+        public void TestM[||]ethod()
+        {
+            System.Console.WriteLine(""Hello World"");
+        }
+    }
+}";
+            var propertyTest1 = @"
+using System;
+namespace PushUpTest
+{
+    interface IInterface
+    {
+        int TestProperty { get; }
+    }
+
+    public class TestClass : IInterface
+    {
+        public int TestPr[||]operty { get; private set; }
+    }
+}";
+
+            var propertyTest2 = @"
+using System;
+namespace PushUpTest
+{
+    interface IInterface
+    {
+        int TestProperty { get; set; }
+    }
+
+    public class TestClass : IInterface
+    {
+        public int TestPr[||]operty { get; set; }
+    }
+}";
+
+            var eventTest = @"
+using System;
+namespace PushUpTest
+{
+    interface IInterface
+    {
+        event EventHandler Event2;
+    }
+
+    public class TestClass : IInterface
+    {
+        public event EventHandler Event1, Eve[||]nt2, Event3;
+    }
+}";
+            await TestNoRefactoringProvidedAsync(methodTest);
+            await TestNoRefactoringProvidedAsync(propertyTest1);
+            await TestNoRefactoringProvidedAsync(propertyTest2);
+            await TestNoRefactoringProvidedAsync(eventTest);
```

Comment:
in general, it's nice to just have these be separate test methods.  otherwise you can be in the situation where something fails, you fix it, then you rerun and the test still fails, etc. etc. Having each test one scenario is nice and simple and consistent.

Reply:
OK